.github/workflows/pulumi-up.yml

```diff
@@ -0,0 +1,40 @@
+name: Pulumi Up
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  pulumi-preview:
+    name: Pulumi Preview
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+
+      - name: Run Pulumi Up
+        uses: pulumi/actions@v6
+        with:
+          command: up
```

Comment:
Where do I read the documentation for this command? I found this, https://github.com/pulumi/actions . It doesn't quite explain what it does.

Reply:
This is probably the best place to start:

https://www.pulumi.com/docs/iac/cli/commands/pulumi_up/

`pulumi up` is like `terraform up`. It executes all the changes in state.